src/services/ChallengeService.js

```diff
@@ -2282,6 +2282,9 @@ function sanitizeData(data, challenge) {
  */
 async function deleteChallenge(currentUser, challengeId) {
   const challenge = await challengeDomain.lookup(getLookupCriteria("id", challengeId));
```

Comment:
Let's use a scanCriteria here with

1. id: challengeId
2. status: New

We don't need to fetch a challenge that's not New.